tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -1932,6 +1966,44 @@ public:
    ~RField() override = default;
 };
 
+template <typename ItemT>
+class RField<std::set<ItemT>> : public RSetField {
+   using ContainerT = typename std::set<ItemT>;
+
+public:
+   static std::string TypeName() { return "std::set<" + RField<ItemT>::TypeName() + ">"; }
+
+   RField(std::string_view name, std::unique_ptr<Detail::RFieldBase> itemField) : RSetField(name, std::move(itemField))
+   {
+   }
```

Comment:
I don't think we need / should have this constructor. If we need it, we need to check that `itemField` is of the correct type.